docs/modules/demos/pages/jupyterhub-keycloak.adoc

```diff
@@ -0,0 +1,197 @@
+A successful login will redirect you back to JupyterHub where different profiles are listed (the drop-down options are visible when you click on the respective fields):
+
+image::jupyterhub-keycloak/server-options.png[]
+
+The explorer window on the left includes HDFS configuration files and a notebook that is already mounted.
+Open the `hdfs` folder where the `core-site.xml` and `hdfs-site.xml` from the discovery ConfigMap of the HDFS cluster are located.
+You can see how to reference the HDFS cluster when reading from it in the notebook (`hdfs://hdfs/`):
+
+image::jupyterhub-keycloak/hdfs-core-site.png[]
+
+Double-click on the file `notebook/process-s3.ipynb`:
+
+image::jupyterhub-keycloak/load-nb.png[]
+
+Run the notebook by selecting "Run All Cells" from the menu:
+
+image::jupyterhub-keycloak/run-nb.png[]
+
+The notebook includes some comments regarding image compatibility and uses a custom image built off the official Spark official that matches the Spark version used in the notebook.
+The java versions also match exactly.
+Python versions need to match at the `major:minor` level, which is why Python 3.11 is used in the custom image.
+
+Once the spark executor has been started (we have specified `spark.executor.instances` = 1) it will spin up as an extra pod.
+We have named the spark job to incorporate the current user (justin-martin).
+JupyterHub has started a pod for the user's notebook instance (`jupyter-justin-martin---bdd3b4a1`) and another one for the spark executor (`process-s3-jupyter-justin-martin-bdd3b4a1-9e9da995473f481f-exec-1`):
+
+[source,console]
+----
+12:49 $ kubectl get pods
+NAME                                   READY   STATUS      RESTARTS   AGE
+...
+jupyter-justin-martin---bdd3b4a1       1/1     Running     0          17m
+process-s3-jupyter-justin-martin-...   1/1     Running     0          2m9s
+...
+----
+
+Stop the kernel in the notebook (which will shut down the spark session and thus the executor) and log out as the current user.
+Log in now as `daniel.king` and then again `isla.williams` (you may need so do this in a clean browser sessions so that existing login cookies are removed).
+This user has been defined as an admin user in the jupyterhub configuration:
+
+[source,yaml]
+----
+  ...
+  hub:
+    config:
+      Authenticator:
+        # don't filter here: delegate to Keycloak
+        allow_all: True
+        admin_users:
+          - isla.williams
+  ...
+----
+
+You should now see user-specific pods for all three users:
+
+
+[source,console]
+----
+16:16 $ kubectl get pods
+NAME                               READY   STATUS      RESTARTS   AGE
+...
+jupyter-daniel-king---181a80ce     1/1     Running     0          6m17s
+jupyter-isla-williams---14730816   1/1     Running     0          4m50s
+jupyter-justin-martin---bdd3b4a1   1/1     Running     0          3h47m
+...
+----
+
+The admin user (`isla.williams`) will also have an extra Admin tab in the JupyterHub console where current users can be managed:
```

Comment:
Needed to look for that tab, could we add the ref `/hub/admin#/` or `http://172.18.0.2:31095/hub/admin#/` here?

Reply:
Done: 80bd2c6